struct.go

```diff
@@ -205,6 +205,20 @@ func setWithProperType(t reflect.Type, key *Key, field reflect.Value, delim stri
 		field.Set(reflect.ValueOf(timeVal))
 	case reflect.Slice:
 		return setSliceWithProperType(key, field, delim, allowShadow, isStrict)
+	case reflect.Ptr:
+		isBoolPtr := t.Elem().Kind() == reflect.Bool
+		if isBoolPtr {
+			boolVal, err := key.Bool()
+			if err != nil {
+				return wrapStrictError(err, isStrict)
+			}
+
+			v := reflect.New(t.Elem())
+			v.Elem().SetBool(boolVal)
+			field.Set(v)
+		} else {
+			return fmt.Errorf("unsupported type '%s'", t)
+		}
```

Comment:
I think `reflect.Ptr` is basically no-type... it is OK we only support `*bool` right now. But for future support of other types, it would be better we use type switch here:

```go
switch t.Elem().Kind() {
	case reflect.Bool:
	default:
}
```

What do you think?

Reply:
Agreed.